### PR TITLE
Feature/Issue 376 Update EnigmaReader Data Connector

### PR DESCRIPTION
**Pull request by hydrosquall**
Implements #376, and includes

- Updating `EnigmaReader`
- Updating tests
- Updating docs
- Updating the main `data.py` library.

The exact sunsetting date of `app.enigma.io` is not determined, but is coming soon.

### CODE REVIEW

**Comment by hydrosquall** [comment]
@jreback , this is ready for review :)

File: docs/source/remote_data.rst [hunk 1]
```diff
@@ -202,15 +202,18 @@ Available expiry dates can be accessed from the ``expiry_dates`` property.
 Enigma
 ======
 
-Access datasets from `Enigma <https://app.enigma.io>`__,
```
Comment (jreback) [inline docs/source/remote_data.rst]: add a note/warning that this changed in 0.6.0

File: docs/source/remote_data.rst [hunk 2]
```diff
@@ -202,15 +202,18 @@ Available expiry dates can be accessed from the ``expiry_dates`` property.
...
+Access datasets from `Enigma <https://public.enigma.com>`__,
 the world's largest repository of structured public data.
 
+Datasets are unique identified by the ``uuid4`` at the end of a dataset's web address.
+For example, the following code downloads USDA Food Recall Data from https://public.enigma.com/datasets/292129b0-1275-44c8-a6a3-2a0881f24fe1.
```
Comment (jreback) [inline docs/source/remote_data.rst]: your https won't render, do like: `` `USDA Food Recal Data <https:.......>`__ ``

File: docs/source/remote_data.rst (hunk 1 above)
Comment (hydrosquall) [inline docs/source/remote_data.rst]: Updated on line 207

File: docs/source/remote_data.rst (hunk 2 above)
Comment (hydrosquall) [inline docs/source/remote_data.rst]: Done :)!

**Comment by hydrosquall** [comment]
@jreback , changes are complete! Let me know if you need anything else tweaked

**Comment by gliptak** [comment]
@jreback Would you like to see other updates to this PR? Thanks

**Comment by hydrosquall** [comment]
@davidastephens - would it be OK to role this PR into the next release?

**Comment by hydrosquall** [comment]
Update: `app.enigma.io` has just been sunset, and points to `public.enigma.com` instead. As a result, the old `EnigmaDatareader` has stopped working, but the new EnigmaDatareader in this PR fixes that.

**Comment by davidastephens** [comment]
Thanks @hydrosquall

**Pull request merged**